connector/src/main/java/org/jboss/as/connector/subsystems/datasources/DataSourceDisable.java

```diff
@@ -67,133 +50,8 @@ public void execute(OperationContext context, ModelNode operation) {
-                                    }
-                                } else {
-                                    throw new OperationFailedException(ConnectorLogger.ROOT_LOGGER.serviceNotAvailable("Data-source.connectionProperty", name));
-                                }
-                            }
-                            if (xaDataSourceConfigServiceName.append("xa-datasource-properties").isParentOf(name)) {
-                                final ServiceController<?> xaConfigPropertyController = registry.getService(name);
-
-                                if (xaConfigPropertyController != null) {
-                                    if (ServiceController.State.UP.equals(xaConfigPropertyController.getState())) {
-                                        xaConfigPropertyController.setMode(ServiceController.Mode.NEVER);
-                                    } else {
-                                        throw new OperationFailedException(ConnectorLogger.ROOT_LOGGER.serviceAlreadyStarted("Data-source.xa-config-property", name));
-                                    }
-                                } else {
-                                    throw new OperationFailedException(ConnectorLogger.ROOT_LOGGER.serviceNotAvailable("Data-source.xa-config-property", name));
-                                }
-                            }
-                        }
-
-
-                        if (xaDataSourceConfigController != null) {
-                            context.removeService(xaDataSourceConfigController);
-                        }
-
-                        if (dataSourceConfigController != null) {
-                            context.removeService(dataSourceConfigController);
-                        }
-
-                        context.completeStep(new OperationContext.RollbackHandler() {
-                            @Override
-                            public void handleRollback(OperationContext context, ModelNode operation) {
-                                try {
-                                    reEnable(context, operation, datasourceRegistration);
-                                } catch (OperationFailedException e) {
-                                    // ignored
-                                }
-                            }
-                        });
-                    }
-                }, OperationContext.Stage.RUNTIME);
-            } else {
-                context.addStep(new OperationStepHandler() {
-                    @Override
-                    public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
-                        context.reloadRequired();
-                        context.completeStep(OperationContext.RollbackHandler.REVERT_RELOAD_REQUIRED_ROLLBACK_HANDLER);
-                    }
-                }, OperationContext.Stage.RUNTIME);
-            }
-        }
-        context.stepCompleted();
```

Comment:
WFLY-4294 says the op will fail if run on a server. Which is the right behavior for an op like this that we are maintaining only for use on legacy hosts in a domain. So then, you'd need:

```
if (context.isNormalServer()) {
    throw new OperationFailedException(....);
}
```